src/momento/requests/vector_index/item.py

```diff
@@ -19,19 +19,29 @@ class Item:
     vector: list[float]
     """The vector of the item."""
 
-    metadata: dict[str, str] = field(default_factory=dict)
+    metadata: dict[str, str | int | float | bool | list[str]] = field(default_factory=dict)
```

Comment:
I left the types enumerated like this because the IDE (VSCode, PyCharm) will not expand type aliases in an overlay. I can extract it to a type alias if you feel strongly about it.